src/sdltiles.cpp

```diff
@@ -3690,45 +3708,60 @@ void rescale_tileset( int size )
     game_ui::init_ui();
 }
 
-cata::optional<tripoint> input_context::get_coordinates( const catacurses::window &capture_win_ )
+window_dimensions get_window_dimensions( const catacurses::window &win )
 {
-    if( !coordinate_input_received ) {
-        return cata::nullopt;
-    }
+    cata_cursesport::WINDOW *const pwin = win.get<cata_cursesport::WINDOW>();
 
-    cata_cursesport::WINDOW *const capture_win = ( capture_win_.get() ? capture_win_ :
-            g->w_terrain ).get<cata_cursesport::WINDOW>();
-
-    // this contains the font dimensions of the capture_win,
-    // not necessarily the global standard font dimensions.
-    int fw = fontwidth;
-    int fh = fontheight;
-    // tiles might have different dimensions than standard font
-    if( use_tiles && capture_win == g->w_terrain ) {
-        fw = tilecontext->get_tile_width();
-        fh = tilecontext->get_tile_height();
-        // add_msg( m_info, "tile map fw %d fh %d", fw, fh);
-    } else if( map_font && capture_win == g->w_terrain ) {
+    window_dimensions dim;
+    if( use_tiles && win == g->w_terrain ) {
```

Comment:
Apparently sometimes `g` could be empty when there are errors (see #39258), so we need to check for non-emptiness prior to comparing `win` with members of `g`, e.g.:

```CPP
    window_dimensions dim;
    dim.scaled_font_size.x = fontwidth;
    dim.scaled_font_size.y = fontheight;
    if( g ) {
        if( use_tiles && win == g->w_terrain ) {
            // tiles might have different dimensions than standard font
            dim.scaled_font_size.x = tilecontext->get_tile_width();
            dim.scaled_font_size.y = tilecontext->get_tile_height();
        } else if( map_font && win == g->w_terrain ) {
            // map font (if any) might differ from standard font
            dim.scaled_font_size.x = map_font->fontwidth;
            dim.scaled_font_size.y = map_font->fontheight;
        } else if( overmap_font && win == g->w_overmap ) {
            dim.scaled_font_size.x = overmap_font->fontwidth;
            dim.scaled_font_size.y = overmap_font->fontheight;
        }
    }
```